cmd/antrea-agent/agent.go

```diff
@@ -127,7 +128,16 @@ func run(o *Options) error {
 
 	go agentMonitor.Run(stopCh)
 
+	serverOpt, err := antctl.NewServerOptions(antctl.Definition, agentMonitor, nil, antctl.DefaultAgentLocalPort)
+	if err != nil {
+		return fmt.Errorf("error initializing antctl server: %w", err)
+	}
+	serverStoppedCh := antctl.StartLocalServer(serverOpt, stopCh)
+
 	<-stopCh
+
 	klog.Info("Stopping Antrea agent")
+	<-serverStoppedCh // check if the antctl server stopped
```

Comment:
I think that shutting down LocalServer by adding a defer clause here can avoid introducing serverStoppedCh here.

Reply:
`serverStoppedCh` here is basically a `Future` like stuff. Since the agent will definitely be killed by the second signal, I use the channel here to ensure the server does have a `graceful` stop in best efforts.

Reply:
Any reason that the server cannot be the same as the others (cniServer, nodeRouteController, networkPolicyController and agentMonitor)? Can we initialize it first, then run it in a separate go routine?

Reply:
OK, I removed the wait of `stoppedCh` and we may need to discuss how to uniform them in future.